### PR TITLE
GH-2124 Add admin commands Logouts and Assists

### LINKED ISSUE #2124: Nowe komendy admina logout, asysty
Komendy admina do resetowania logout'ów i asyst
/ga logouts [nick] [logouts]
/ga assists [nick] [assists]

### PR DESCRIPTION
**Pull request by the-codeboy**
resolves #2124
I used google translate for the messages so they're probably not that good

### CODE REVIEW

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/MessageConfiguration.java [hunk 1]
```diff
@@ -619,6 +619,16 @@ public class MessageConfiguration extends OkaeriConfig {
     @Comment("Dostępne zmienne: {GUILD}, {LIVES}")
     public String adminLivesChanged = "&aUstawiono &7{LIVES} &azyc dla gildii &7{GUILD}&a!";
 
+    public String adminNoLogoutsGiven = "&cWpisz liczbę wylogowań!";
+
+    @Comment("Dostępne zmienne: {PLAYER}, {LOGOUTS}")
+    public String adminLogoutsChanged = "&aUstawiono &7{LOGOUTS} &awylogowań dla gracza &7{PLAYER}";
+
+    public String adminNoAssistsGiven = "&cWpisz liczbę asystuje!";
```
Comment (CDFN) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/MessageConfiguration.java:627]:
```suggestion
    public String adminNoAssistsGiven = "&cWpisz liczbę asyst!";
```

**Review by CDFN — CHANGES_REQUESTED** [review]

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/MessageConfiguration.java [hunk 2]
```diff
@@ -619,6 +619,16 @@ public class MessageConfiguration extends OkaeriConfig {
...
+
+    @Comment("Dostępne zmienne: {PLAYER}, {ASSISTS}")
+    public String adminAssistsChanged = "&aUstawiono &7{ASSISTS} &aasystuje dla gracza &7{PLAYER}";
```
Comment (CDFN) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/MessageConfiguration.java:630]:
```suggestion
    public String adminAssistsChanged = "&aUstawiono &7{ASSISTS} &aasyst dla gracza &7{PLAYER}";
```

**Comment by CDFN** [comment]
As you can see build has failed. You're missing imports for classes `Assists` & `Logouts`.

**Comment by the-codeboy** [comment]
This should be fixed now. Also I noticed that normally the commands have Command in their class name so I changed that for Assists and Logouts

**Comment by the-codeboy** [comment]
So the imports should be sorted alphabetically?

**Comment by P3ridot** [comment]
> So the imports should be sorted alphabetically?

Yes

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/MessageConfiguration.java [hunk 3]
```diff
@@ -619,6 +619,16 @@ public class MessageConfiguration extends OkaeriConfig {
     @Comment("Dostępne zmienne: {GUILD}, {LIVES}")
     public String adminLivesChanged = "&aUstawiono &7{LIVES} &azyc dla gildii &7{GUILD}&a!";
 
+    public String adminNoLogoutsGiven = "&cWpisz liczbę wylogowań!";
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/MessageConfiguration.java:622]: `Wpisz` -> `Podaj` like in other commands messages

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/MessageConfiguration.java [hunk 4]
```diff
@@ -619,6 +619,16 @@ public class MessageConfiguration extends OkaeriConfig {
...
+
+    @Comment("Dostępne zmienne: {PLAYER}, {LOGOUTS}")
+    public String adminLogoutsChanged = "&aUstawiono &7{LOGOUTS} &awylogowań dla gracza &7{PLAYER}";
+
+    public String adminNoAssistsGiven = "&cWpisz liczbę asyst!";
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/MessageConfiguration.java:627]: `Wpisz` -> `Podaj` like in other commands messages

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/MessageConfiguration.java (hunk 3 above)
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/MessageConfiguration.java:622]: Also move logouts and assists messages higer (under adminDeathsChanged)

File: plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/admin/AssistsCommand.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    )
+    public void execute(CommandSender sender, String[] args) {
+        when(args.length < 1, this.messages.generalNoTagGiven);
+        when(args.length < 2, this.messages.adminNoAssistsGiven);
+
+        User user = UserValidation.requireUserByName(args[0]);
+
+        int assists = Integer.parseInt(args[1]);
+
+        user.getRank().setAssists(assists);
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/admin/AssistsCommand.java:28]: Call AssistsChangeEvent (see how it should look for [deaths command](https://github.com/FunnyGuilds/FunnyGuilds/blob/f33f96560f9fc22c9122373a2b4488fe8f613dfa/plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/admin/DeathsCommand.java#L39))

File: plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/admin/LogoutsCommand.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    )
+    public void execute(CommandSender sender, String[] args) {
+        when(args.length < 1, this.messages.generalNoTagGiven);
+        when(args.length < 2, this.messages.adminNoLogoutsGiven);
+
+        User user = UserValidation.requireUserByName(args[0]);
+
+        int logouts = Integer.parseInt(args[1]);
+
+        user.getRank().setLogouts(logouts);
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/admin/LogoutsCommand.java:28]: Call LogoutsChangeEvent (see how it should look for [deaths command](https://github.com/FunnyGuilds/FunnyGuilds/blob/f33f96560f9fc22c9122373a2b4488fe8f613dfa/plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/admin/DeathsCommand.java#L39))

File: plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/FunnyCommandsConfiguration.java [hunk 7]
```diff
@@ -134,7 +136,9 @@ public static FunnyCommands createFunnyCommands(FunnyGuilds plugin) {
                 .command("tag", commands.admin.tag, TagCommand.class)
                 .command("teleport", commands.admin.teleport, TeleportCommand.class)
                 .command("unban", commands.admin.unban, UnbanCommand.class)
-                .command("validity", commands.admin.validity, ValidityAdminCommand.class);
+                .command("validity", commands.admin.validity, ValidityAdminCommand.class)
+                .command("logouts", commands.admin.logouts, LogoutsCommand.class)
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/FunnyCommandsConfiguration.java:140]: Command here are in alphabetical order so do the same for new commands

**Review by P3ridot — CHANGES_REQUESTED** [review]

**Review by dzikoysk — APPROVED** [review]

**Review by P3ridot — APPROVED** [review]

**Comment by dzikoysk** [comment]
Thanks!

**Pull request merged**